cmd/cmd.go

```diff
@@ -63,3 +72,16 @@ func Execute(rootCmd fw.Command) {
 		os.Exit(1)
 	}
 }
+
+// listenToSystemSignals handles OS signals
+func listenToSystemSignals(cancelFn context.CancelFunc, onInterrupt func()) {
+	signalChan := make(chan os.Signal, 1)
+	// listen to Interrupt
```

Comment:
We may want to consider replacing this comment with a short version of:

**SIGINT** is the interrupt signal. The terminal sends it to the foreground process when the user presses ctrl-c. The default behavior is to terminate the process, but it can be caught or ignored. The intention is to provide a mechanism for an orderly, graceful shutdown.

**SIGQUIT** is the dump core signal. The terminal sends it to the foreground process when the user presses ctrl-\. The default behavior is to terminate the process and dump core, but it can be caught or ignored. The intention is to provide a mechanism for the user to abort the process. You can look at SIGINT as "user-initiated happy termination" and SIGQUIT as "user-initiated unhappy termination."

**SIGTERM** is the termination signal. The default behavior is to terminate the process, but it also can be caught or ignored. The intention is to kill the process, gracefully or not, but to first allow it a chance to cleanup.

**SIGKILL** is the kill signal. The only behavior is to kill the process, immediately. As the process cannot catch the signal, it cannot cleanup, and thus this is a signal of last resort.

**SIGSTOP** is the pause signal. The only behavior is to pause the process; the signal cannot be caught or ignored. The shell uses pausing (and its counterpart, resuming via SIGCONT) to implement job control.

So that the readers have full context of the design decisions we made here.